src/Tree/HydroTree.cpp

```diff
@@ -686,11 +710,37 @@ void HydroTree<ndim,ParticleType>::UpdateTimestepsLimitsFromDistantParticles
       if (Nactive == 0) continue ;
 
       // Make local copies of active particles & update vsig_max
-      for (j=0; j<Nactive; j++) activepart[j] = partdata[activelist[j]];
+      for (int j=0; j<Nactive; j++) activepart[j] = partdata[activelist[j]];
 
       tree->ComputeSignalVelocityFromDistantInteractions(cell, Nactive, activepart, partdata);
 
-      for (j=0; j<Nactive; j++) {
+#ifdef MPI_PARALLEL
+      if (!only_imported_particles) {
+		  // Loop over pruned trees as well
+		  // Doing it after the local tree walk is more efficient as vsig_max might have been already increased
+		  // TODO: is it more efficient to have this loop inside or outside the one over particles? In this way we are trashing
```

Comment:
I'd like input on this comment. It's probably more efficient to swap the loop over the pruned trees with the loop over active cells (locality of the tree in the cache is far more important than the locality of the cell). Should we bother? It would be a problem for openmp (would you parallelize over trees at this point? it's a waste if there are more openmp threads than mpi jobs). And it's a pattern we are using also in other places, even if it's not optimal.